frontend/src/components/NavBar.jsx

```diff
@@ -1,21 +1,31 @@
-import { useContext, useEffect } from "react";
+import { useContext, useEffect, useState } from "react";
 import { Sun, MoonStarsFill } from "react-bootstrap-icons";
 import { Link, NavLink, useLocation } from "react-router-dom";
 
 import { ThemeContext } from "../context/ThemeContext";
 import { BowtieVersionContext } from "../context/BowtieVersionContext";
+import { dialectToName } from "..";
 
 const NavBar = () => {
+  const [isDialectPage, setIsDialectPage] = useState(false);
   const { isDarkMode, toggleDarkMode } = useContext(ThemeContext);
   const { version } = useContext(BowtieVersionContext);
-  const { hash, key } = useLocation();
+  const { pathname, hash, key } = useLocation();
 
   useEffect(() => {
     if (hash) {
       const targetElement = document.getElementById(hash.substring(1));
       targetElement?.scrollIntoView({ behavior: "smooth" });
     }
-  }, [key, hash]);
+    const matchPath = () => {
```

Comment:
You could get rid of all this logic to compute `isDialectPage` and use other router APIs, something lime this (outside of `useEffect` of course):
```js
  const rootMatch = useMatch('/')
  const dialectsMatch = useMatch('/dialects/*')
  const isDialectPage = matches1 || matches2
```